src/gui/widgets/multiline_text.hpp

```diff
@@ -0,0 +1,391 @@
+/*
+	Copyright (C) 2008 - 2024
+	by babaissarkar(Subhraman Sarkar) <suvrax@gmail.com>
+	Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+	This program is free software; you can redistribute it and/or modify
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY.
+
+	See the COPYING file for more details.
+*/
+
+#pragma once
+
+#include "gui/widgets/text_box_base.hpp"
+#include "gui/widgets/text_box.hpp"
+#include "gui/widgets/window.hpp"
+#include "gui/widgets/scrollbar_container.hpp"
+
+namespace gui2
+{
+namespace implementation
+{
+struct builder_multiline_text;
+}
+
+// ------------ WIDGET -----------{
+
+/**
+ * @ingroup GUIWidgetWML
+ *
+ * Base class for a multiline text area.
+ *
+ * The resolution for a text box also contains the following keys:
+ * Key          |Type                                    |Default  |Description
+ * -------------|----------------------------------------|---------|-----------
+ * text_x_offset| @ref guivartype_f_unsigned "f_unsigned"|""       |The x offset of the text in the text box. This is needed for the code to determine where in the text the mouse clicks, so it can set the cursor properly.
+ * text_y_offset| @ref guivartype_f_unsigned "f_unsigned"|""       |The y offset of the text in the text box.
+ * The following states exist:
+ * * state_enabled - the text box is enabled.
+ * * state_disabled - the text box is disabled.
+ * * state_focussed - the text box has the focus of the keyboard.
+ * The following variables exist:
+ * Key          |Type                                |Default  |Description
+ * -------------|------------------------------------|---------|-----------
+ * label        | @ref guivartype_t_string "t_string"|""       |The initial text of the text box.
+ * history      | @ref guivartype_string "string"    |""       |The name of the history for the text box. A history saves the data entered in a text box between the games. With the up and down arrow it can be accessed. To create a new history item just add a new unique name for this field and the engine will handle the rest.
+ */
```

Comment:
Add "editable"?

Reply:
Added in 8e4285e1e0564579751f1e9b8846594c3e426159.